packages/mui-material/src/Button/Button.js

```diff
@@ -86,7 +86,7 @@ const ButtonRoot = styled(ButtonBase, {
       ownerState.fullWidth && styles.fullWidth,
     ];
   },
-})(({ theme }) => {
+})(styled.pure(theme => {
```

Comment:
I think Pigment CSS needs to support this too.

Reply:
Let's wait @brijesh to confirm if this is possible

Reply:
Let's continue the discussion here: https://github.com/mui/material-ui/pull/43372#discussion_r1722789324